pennylane/measurements/sample.py

```diff
@@ -195,7 +195,7 @@ def numeric_type(self):
             # Computational basis samples
             return int
         int_eigval_obs = {qml.X, qml.Y, qml.Z, qml.Hadamard, qml.Identity}
-        tensor_terms = self.obs.obs if hasattr(self.obs, "obs") else [self.obs]
+        tensor_terms = self.obs.obs if isinstance(self.obs, qml.operation.Tensor) else [self.obs]
```

Comment:
What about `Prod`?

Reply:
So `Prod` is already handled without accessing the individual terms. The problem here is that if we try to also obtain `obs` from the `Prod` instance it is a) unnecessary and b) raises the deprecation warning (and therefore fails a related test)